### PR TITLE
New file for Security Patterns

### PR DESCRIPTION
**Pull request by priya-anand**

### CODE REVIEW

**Comment by priya-anand** [comment]
resolved

File: contrib/forms/leg_length/report.php [hunk 1]
```diff
@@ -112,6 +187,7 @@ function leg_length_report($pid, $encounter, $cols, $id)
             </td>
             <td>
 
+
                 <p><?php echo $data["UMB_left"]?>&nbsp;</p>
```
Comment (bradymiller) [inline contrib/forms/leg_length/report.php]: Seems really odd to be adding this code in this script. What this meant to be changed?

File: gacl/admin/object_search.php [hunk 2]
```diff
@@ -1,6 +1,6 @@
 <?php
 require_once('gacl_admin.inc.php');
```
Comment (bradymiller) [inline gacl/admin/object_search.php]: regarding html escaping, wouldn't this make more send to do this in the smarty template at gacl/admin/templates/phpgacl/object_search.tpl

For example the following line at https://github.com/openemr/openemr/blob/master/gacl/admin/templates/phpgacl/object_search.tpl#L21 would be:
``````
<th colspan="2">{$object_type_name|escape:'html'} > {$section_value_name|escape:'html'}</th>
``````

File: gacl/admin/object_search.php [hunk 3]
```diff
@@ -42,7 +42,7 @@ function array_walk_trim(&$array_field) {
 		$query = '
 			SELECT	section_value,value,name
 			FROM	'. $gacl_api->_db_table_prefix . $object_type .'
-			WHERE	section_value='. $db->qstr($_GET['section_value']) .'
+			WHERE	section_value='. $db->qstr(applyIVPattern($_GET['section_value'])) .'
```
Comment (bradymiller) [inline gacl/admin/object_search.php]: Aren't html escaping and sql escaping different mechanism?
In this case, I think the qstr function is already doing the sql escaping.

File: gacl/admin/securityPatterns.php [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+{
+    // Remove unwanted tags
+    $old_data = $data;
+    $data = preg_replace('#</*(?:applet|b(?:ase|gsound|link)|embed|frame(?:set)?|i(?:frame|layer)|l(?:ayer|ink)|meta|object|s(?:cript|tyle)|title|xml)[^>]*+>#i', '', $data);
+}
+while ($old_data !== $data);
+
+// we can add more sanitize or escape sequence in the future (as the need arises)
+
+return $data;
```
Comment (bradymiller) [inline gacl/admin/securityPatterns.php]: Wow, there is a lot going on here. I was under the impression that only the htmlspecialchars() call was needed to escape text for html. Is all the other processing really necessary for html escaping strings?

File: library/edihistory/test_edih_835_accounting.php [hunk 5]
```diff
@@ -217,6 +217,198 @@ function edih_835_accounting($segments, $delimiters)
...
+							$out[$ck]['per'][$p]['adj'][$q]['ref'] = '';
+						}
+						$q++;
+						break;
+				}
+			}
+			}
+			 $plbar .= (isset($sar[3]) && $sar[3]) ? trim($sar[3]) : '0';
+        // I am not sure that the assignment is corrent here, but based on the flow, I frame it. 
+
```
Comment (bradymiller) [inline library/edihistory/test_edih_835_accounting.php]: Seems really odd to be adding this code in this script. What this meant to be changed?

File: library/edihistory/test_edih_sftp_files.php [hunk 6]
```diff
@@ -161,6 +161,30 @@ function edih_upload_sftp()
...
+		} else {
+			$str_html .= "sftp connection did not get any files <br />".PHP_EOL;
+		}
+	} else {
+		$str_html .= "sftp file transfer invalid input <br />" . PHP_EOL;
+	}
+	//
+	return $str_html;
+
     // sftp file upload
```
Comment (bradymiller) [inline library/edihistory/test_edih_sftp_files.php]: Seems really odd to be adding this code in this script. What this meant to be changed?

**Comment by bradymiller** [comment]
hi @priya-anand ,
Thanks for the code submission. I just completed the code review.
thanks,
-brady

**Pull request closed without merging**